### PR TITLE
feat: add support for configuring named connections

### LINKED ISSUE #273: Allow callers to configure multiple "connectors" per socket factory
### Feature Description

This is a feature request to add support for a "Connector" type that can be registered with the SocketFactory.

In real usage, an application would first call `ConnectorRegistry.register()` to register the configuration for a named connection.

``` java
ConnectorConfig config =  new ConnectorConfig.Builder()
            .withTargetPrincipal("example@project.iam.googleapis.com")
            .withDelegates(Arrays.asList("delegate@project.iam.googleapis.com"))
            .build();
ConnectorRegistry.register("my-connector", config);
```

Then the application would use the connector by adding the `alloydbNamedConnector` property:

``` java
HikariConfig config = new HikariConfig();

config.setJdbcUrl("jdbc:postgresql:///postgres");
config.setUsername(System.getenv("ALLOYDB_USER"));
config.setPassword(System.getenv("ALLOYDB_PASS"));

config.addDataSourceProperty("socketFactory", "com.google.cloud.alloydb.SocketFactory");
config.addDataSourceProperty("alloydbInstanceName", System.getenv("ALLOYDB_INSTANCE_NAME"));
config.addDataSourceProperty("alloydbNamedConnector", "my-connector");

HikariDataSource dataSource = new HikariDataSource(config);
```

The application can stop individual named connectors, or all connectors:

``` java
ConnectorRegistry.close("my-connector");
```

The application can shutdown the connector, preventing any new connections and stoping all connector threads.

``` java
ConnectorRegistry.shutdown();
```

This is porting https://github.com/GoogleCloudPlatform/cloud-sql-jdbc-socket-factory/issues/1226 to alloydb-java-connector.

### PR DESCRIPTION
**Pull request by ttosta-google**
Fixes #273

### CODE REVIEW

File: alloydb-jdbc-connector/src/main/java/com/google/cloud/alloydb/ConnectionConfig.java [hunk 1]
```diff
@@ -17,21 +17,18 @@
...
 import com.google.cloud.alloydb.v1.InstanceName;
-import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
 import java.util.Objects;
 import java.util.Properties;
-import java.util.stream.Collectors;
 
 public class ConnectionConfig {
```
Comment (hessjcg) [inline alloydb-jdbc-connector/src/main/java/com/google/cloud/alloydb/ConnectionConfig.java:26]: This class should be package-private, not public.

**Review by hessjcg — COMMENTED** [review]
Looks like a faithful port of the Java implementation. Nice work. 
One minor comment about package scope. Otherwise LGTM.

File: alloydb-jdbc-connector/src/main/java/com/google/cloud/alloydb/ConnectionConfig.java (hunk 1 above)
Comment (ttosta-google) [reply alloydb-jdbc-connector/src/main/java/com/google/cloud/alloydb/ConnectionConfig.java]: Updated. Thanks for catching that.

**Review by hessjcg — APPROVED** [review]
LGTM

**Pull request merged**